### ISSUE TITLE
Cannot push to private/owned repo. "Authentication failed"

### ISSUE THREAD

**Issue by reconstrukt** [desktop/desktop#10166]
### Describe the bug

Attempting to push to a private repo I own, but getting this error: 

> Authentication failed. Some common reasons include:
> 
> - You are not logged in to your account: see File > Options.
> - You may need to log out and log back in to refresh your token.
> - You do not have permission to access this repository.
> - The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
> - If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account."

### Version & OS

Github Desktop Version 2.5.3
Windows 10 Pro 64 bit

### Steps to reproduce the behavior

1. Go to Github Desktop
2. Click on "Push Origin"
3. See error

### Expected behavior

Should push successfully.
Have been working on this project repo for several weeks, pushing 10x daily without issue.

### Logs

See attached log
[2020-07-08.desktop.production.log](https://github.com/desktop/desktop/files/4888203/2020-07-08.desktop.production.log)

### Additional context

I can confirm the following:
 - I am logged into Github Desktop
 - Can fetch/pull/push from other private/owned repos in Github Desktop without issue, confirmed this on several repos
 - Not using SSH (that I am aware of?)

I have tried the following:
 - Laptop reboot
 - Logging out of Github desktop, logging back in
 - Re-cloning the repo to another local directory, attempting to push (minor/benign) change - also failed with same error 
 - Completely removing and re-installing Github desktop, and manually re-adding the local repo

**Comment by Luckyfive** [comment]
@reconstrukt try the following steps:

1. Go to my Github account settings and click on Applications
2. Go to Authorized OAuth Apps and revoke the Github Desktop application
3. Follow the Sign out and Sign In steps within the Github Desktop application
4. Issue seems fixed for now; I am able to publish a new branch and any changes

**Comment by Luckyfive** [comment]
I was encountering the same issues by the way: https://github.com/desktop/desktop/issues/10165

**Comment by reconstrukt** [comment]
@Luckyfive yes, tried that. should have added to my OP above.

I revoked the desktop client app via my GH account, logged out of the desktop client, and re-logged in. This re-auth’ed Github Desktop on my account. 

But, still did not fix the problem, still cannot push to this repo.

**Comment by steveward** [comment]
@reconstrukt thanks for the report. Are you able to push to the repository from the command line?

**Comment by reconstrukt** [comment]
@steveward i don't know, but can try. how would i do that?

**Comment by Luckyfive** [comment]
@reconstrukt 

1. git add <file>
2. git commit -m "<message>"
3. git push

**Comment by steveward** [comment]
You can go to `Repository` > `Open in <command_line>` (you may need to install Git for Windows if you have not done so already). Then just run the `git push` command to see if it succeeds.

**Comment by reconstrukt** [comment]
@Luckyfive @steveward 

No dice.  Opened powershell, and 

`cd c:\users\matt\sites\probook`
`git status` shows: 

> On branch master
> Your branch is ahead of 'origin/master' by 1 commit.
>   (use "git push" to publish your local commits) in sync with one change to push 

then running `git push` fails, says:

> Unable to read from remote repo

@steveward also tried using standard command prompt via "Repository -> Open in Command Line"

same result as above

**Comment by reconstrukt** [comment]
So strange. So the command line thing made me think to also try this:

 1) "Repository -> Show in Explorer"
 2) Right-click in Explorer -> selected "Git Bash Here" from context menu
 3) In git bash shell, run `git push` 
 4) Holy moly, it worked

... and now, after that, I can also push from Git Desktop client again ...

??

**Comment by steveward** [comment]
@reconstrukt I'm not sure what would have caused that to get things working in GitHub Desktop, but I'm glad to hear it seems to be working again. Let us know if you run into this again.

**Comment by reconstrukt** [comment]
@steveward @Luckyfive 

IDK if this is related or helpful, but for sake of comprehensiveness its worth mentioning: 

I noticed that in the logs the error seemed to be related to 
> Could not read from remote repository

I thought perhaps this was network-related, so I cleared the DNS cache, 
> ipconfig /flushdns

I didn't retest after that (bad QAer!) but then went right to my next hunch which was to try a git push via bash shell, outlined here: https://github.com/desktop/desktop/issues/10166#issuecomment-655598627

Which worked as mentioned above.

This being network-related doesn't necessarily make logical sense, as I could fetch from this repo just fine, and could fetch/pull/push from other owned/private repos fine too.  But who knows [/shrug]

HTH! Matt

**Comment by MikeLanglois** [comment]
> So strange. So the command line thing made me think to also try this:
> 
> 1. "Repository -> Show in Explorer"
> 2. Right-click in Explorer -> selected "Git Bash Here" from context menu
> 3. In git bash shell, run `git push`
> 4. Holy moly, it worked
> 
> ... and now, after that, I can also push from Git Desktop client again ...
> 
> ??

Apologies for a reply on a closed record, but following this I get the error:
fatal: unable to access '*location removed*': SSL certificate problem: unable to get local issuer certificate

Any ideas on a fix?

**Comment by Luckyfive** [comment]
@MikeLanglois have you tried re-cloning the repo and just copying over your changes to the new folder?

**Comment by MikeLanglois** [comment]
@Luckyfive Thanks for the reply. I did try pushing to a separate repo, however the original issue mentioned here in Github still remains so doesn't solve it.

It seems like the original error message was a red herring for my actual problem which was git not liking an upload of a .suo file. Added to gitignore and problem resolved for me. Will leave this here for anyone who comes across this like I did while googling
